appcenter-data/ios/AppCenterReactNativeData/AppCenterReactNativeData.m

```diff
@@ -103,63 +195,85 @@ + (void)register {
                   writeOptions:(NSDictionary *)writeOptionsMap
                   resolver:(RCTPromiseResolveBlock)resolve
                   rejecter:(RCTPromiseRejectBlock)reject) {
-    MSWriteOptions *writeOptions = [self getWriteOptions:writeOptionsMap];
+    MSWriteOptions *writeOptions = getWriteOptions(writeOptionsMap);
     MSDictionaryDocument *document = [[MSDictionaryDocument alloc] initFromDictionary:documentMap];
     [MSData replaceDocumentWithID:documentID
                          document:document
                         partition:partition
                      writeOptions:writeOptions
-                completionHandler:[self dataCompletionHandler:kMSReplaceFailedErrorCode resolve:resolve reject:reject]];
+                completionHandler:dataCompletionHandler(kMSRemoveFailedErrorCode, resolve, reject)];
 }
 
 RCT_EXPORT_METHOD(remove:(NSString *)documentID
                   partition:(NSString *)partition
-                  writeOptions:(NSDictionary *)writeOptionsMap
+                  writeOptions:(NSDictionary* )writeOptionsMap
                   resolver:(RCTPromiseResolveBlock)resolve
                   rejecter:(RCTPromiseRejectBlock)reject) {
-    MSWriteOptions *writeOptions = [self getWriteOptions:writeOptionsMap];
+    MSWriteOptions *writeOptions = getWriteOptions(writeOptionsMap);
     [MSData deleteDocumentWithID:documentID
                        partition:partition
                     writeOptions:writeOptions
-               completionHandler:[self dataCompletionHandler:kMSRemoveFailedErrorCode resolve:resolve reject:reject]];
+               completionHandler:dataCompletionHandler(kMSRemoveFailedErrorCode, resolve, reject)];
+}
+
+- (void)closeList:(NSString *)paginatedDocumentsId {
```

Comment:
`closeList` naming sounds weird to me.

If it's only removing an item from the `_paginatedDocuments`, why not call `[_paginatedDocuments removeObjectForKey:paginatedDocumentsId];` directly?

Reply:
I remove `closeList`. I just found out that I can call `[self close:paginatedDocumentsId];` instead. I'm still trying to navigate objective-c syntax.